### DISCUSSION TITLE
How can I copy-paste from a VM through VNC ?

### DISCUSSION THREAD

**Discussion by TheCarpetMerchant** [codemagic-ci-cd/codemagic-docs#1946] [Ideas]
I'm trying to copy-paste a file from the Mac Mini M1 VM through VNC, but I can't for some reason.
Right-click -> Copy doesn't allow me to paste on my Windows desktop.

**Comment by kulakarla** [comment]
Hello @TheCarpetMerchant ,

are you copying text or a file?
Copying a file is impossible with our VNC setup currently.
Text cross-copying should be possible, however, there may potentially be tons of issues it. For some people it works, for some it doesn't. We could potentially debug it for days and not get it fixed. I recommend you to use a 3rd party upload service in the VNC connection, e.g [easyupload](https://easyupload.io).

**Reply by TheCarpetMerchant** [reply]
Yeah, that's what I ended up doing. Oh well, it is what it is, will close. Thanks !
